molecule/filter-plugin-collection/converge.yml

```diff
@@ -0,0 +1,10 @@
+---
+- name: Converge
+  hosts: all
+  gather_facts: false
+  roles:
+    - role: confluent.platform.variables
```

Comment:
oh this looks like the distinction

Reply:
but if the current dir is the base of the repo then we could just test
```
- include_role:
    name: confluent.platform.variables
- assert: #whatever 
- include_role:
    name: variables
- assert: #whatever
```

Reply:
IMO that would not be a valid test, becuase then you are importing the same role twice - just using a different name for it (simple and FQN). Ansible will normally not apply the same role twice - unlessed forced to do so.

And if you use the collection, you will not clone the repo yourself. It will be cloned (or downloaded from Ansible Galaxy) by the ansible-galaxy CLI.

Reply:
im not certain ansible will see them as the same role. for this:
```
- include_role:
    name: variables
```
ansible will look in normal role paths for the variables role

and for 
```
- include_role:
    name: confluent.platform.variables
```
It seems to look for `/Users/dbove/.ansible/collections/ansible_collections/confluent/platform/roles` at least on my laptop.
But splitting among two scenarios i suppose doesn't have too big of a footprint at this point. so this is not really that big of a deal

I can replicate this error locally:
```
    TASK [Assert that custom filter plugin works] **********************************
fatal: [instance]: FAILED! => {"msg": "The conditional check 'mds_bootstrap_server_urls == 'http://localhost:8090'' failed. The error was: An unhandled exception occurred while templating '{{mds_http_protocol}}://{{ groups['kafka_broker'] | default(['localhost']) | resolve_hostnames(hostvars) | join(':' + mds_port|string + ',' + mds_http_protocol + '://') }}:{{mds_port}}'. Error was a <class 'ansible.errors.AnsibleError'>, original message: template error while templating string: no filter named 'resolve_hostnames'. String: {{mds_http_protocol}}://{{ groups['kafka_broker'] | default(['localhost']) | resolve_hostnames(hostvars) | join(':' + mds_port|string + ',' + mds_http_protocol + '://') }}:{{mds_port}}"}
```

also in the clone scenario i think you need to add to get it to find the local roles because that one is also erroring out for me
```
provisioner:
  env:
    ANSIBLE_ROLES_PATH: ../../roles/
```

Reply:
Actually thats odd that the clone scenario succeeds in GHA, it does not for me. I'd think it'd have trouble with the roles path bc I get this locally:
```
--> Action: 'converge'
ERROR! the role 'variables' was not found in /Users/dbove/workspace/cp-ansible/molecule/filter-plugin-clone/roles:/Users/dbove/.cache/molecule/cp-ansible/filter-plugin-clone/roles:/Users/dbove/workspace:/Users/dbove/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles:/Users/dbove/workspace/cp-ansible/molecule/filter-plugin-clone
```

Reply:
@domenicbove Which version of molecule are you using? I think they have done quite a few improvements lately. GHA and on my laptop runs the latest version:
```shell
# molecule --version
molecule 3.3.0 using python 3.8
    ansible:2.10.7
    delegated:3.3.0 from molecule
    docker:0.2.4 from molecule_docker
```

Reply:
This has be addressed by upgrading the versions used by the Jenkins build (Ansible and Molecule) and checking out the project to `.ansible/collections/ansible_collections/confluent/platform` - as described in the blog post from Jeff Geerling.